### PR TITLE
Control automatic resubmission on SLURM

### LINKED ISSUE #6389: Disable automatic SLURM Detection
## 🚀 Feature

I want to run my PTL script on a SLURM cluster, but I don't want to use the built in SlurmConnector since I am using Ray to handle distributed training. When using Ray and the SlurmConnector, it results in this error: https://github.com/PyTorchLightning/pytorch-lightning/issues/3651. Is there a way I can disable automatic SLURM detection? Having an environment variable to disable this would really help. Thanks!

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #6389 

Adds the ability to turn on or off the automatic resubmission that Lightning does when a job gets interrupted by the SLURM controller (via signal handling). Users who prefer to let libraries handle the resubmission (e.g. submitit) can now set

```python
trainer = Trainer(plugins=SLURMEnvironment(auto_requeue=False))

``` 
Idea from @ananthsub. 

Whether the the default for `auto_requeue` is True or False is currently being debated on the linked issue. For now, the PR implements the behavior we have already (auto_requeue=True by default).

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda @awaelchli

### CODE REVIEW

File: tests/trainer/connectors/test_signal_connector.py [hunk 1]
```diff
@@ -57,3 +59,30 @@ def training_step(self, batch, batch_idx):
...
+
+        sigusr1_handlers = signal.getsignal(signal.SIGUSR1).signal_handlers
+        assert len(sigusr1_handlers) == 1
+        assert sigusr1_handlers[0].__qualname__ == "SignalConnector.slurm_sigusr1_handler_fn"
+    else:
+        assert signal.getsignal(signal.SIGTERM) is sigterm_handler_default
+        assert signal.getsignal(signal.SIGUSR1) is sigusr1_handler_default
+
+    # restore the signal handlers so the next test can run with system defaults
+    # TODO: should this be done in SignalConnector teardown?
```
Comment (tchaton) [inline tests/trainer/connectors/test_signal_connector.py:86]: Good catch! There should be teardown on the SignalConnector.

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/trainer/connectors/test_signal_connector.py (hunk 1 above)
Comment (awaelchli) [reply tests/trainer/connectors/test_signal_connector.py:86]: yep, working on a follow up

**Review by ananthsub — APPROVED** [review]
Thanks for adding this!

**Comment by marksibrahim** [comment]
Thank you for implementing this!

**Review by carmocca — APPROVED** [review]

**Comment by marksibrahim** [comment]
@awaelchli  Do you know which release this is expected to be a part of? I tried installing the latest 1.5.3 release (from 2 hours ago) and it doesn't seem to include the new SLURM `auto_requeue` argument.

**Comment by carmocca** [comment]
@marksibrahim Lightning 1.6, so a bit far a way. You will have to install directly from the GitHub repository in the meantime.
